### PR TITLE
release 0.10.2

### PR DESCRIPTION
**Pull request by sammlapp**
This release includes bug fixes and minor improvements to 0.10.1. This branch off of develop was created for the 0.10.2 release and does not include breaking changes that will be incorporated in the 0.11.0 release.

### CODE REVIEW

File: opensoundscape/ml/cnn.py [hunk 1]
```diff
@@ -656,9 +666,9 @@ def _train_epoch(self, train_loader, wandb_session=None, progress_bar=True):
                 self._log(f"\tDistLoss: {epoch_loss_avg:.3f}")
 
                 # Evaluate with model's eval function
-                tgts = batch_labels.int().detach().cpu().numpy()
-                scores = logits.int().detach().cpu().numpy()
-                self.eval(tgts, scores, logging_offset=-1)
+                # tgts = batch_labels.detach().cpu().numpy()
+                # scores = logits.detach().cpu().numpy()
+                # self.eval(tgts, scores, logging_offset=-1)
```
Comment (louisfh) [inline opensoundscape/ml/cnn.py:671]: Can we note why these are commented out?

File: opensoundscape/preprocess/actions.py [hunk 2]
```diff
@@ -58,7 +59,11 @@ def set(self, **kwargs):
             f"unexpected arguments: {unmatched_args}. "
             f"The valid arguments and current values are: \n{self.params}"
         )
-        self.params.update(pd.Series(kwargs, dtype=object))
+        # Series.update ignores nan/None values, so we use dictionary.update method
+        new_params = dict(self.params)
+        new_params.update(kwargs)
+        self.params = pd.Series(new_params, dtype=object)
+        # self.params.update(pd.Series(kwargs, dtype=object))
```
Comment (louisfh) [inline opensoundscape/preprocess/actions.py:66]: Note why commented out.

File: opensoundscape/utils.py [hunk 3]
```diff
@@ -204,6 +206,15 @@ def generate_clip_times_df(
     return pd.DataFrame({"start_time": starts, "end_time": ends}).drop_duplicates()
 
 
+def cast_np_to_native(x):
```
Comment (louisfh) [inline opensoundscape/utils.py:209]: No tests for this function. I'm not sure tests would be necessary for a function this simple? But wanted to note.

Comment (louisfh) [reply opensoundscape/utils.py:209]: Maybe just to be on the safe side, the function should raise an error if passed something that is not a np datatype?

File: tests/test_utils.py [hunk 4]
```diff
@@ -168,3 +171,72 @@ def test_make_clip_df_from_label_df(silence_10s_mp3_str, metadata_wav_str):
...
+    rs2 = random.sample(list1000, 100)
+
+    utils.set_seed(input + 1)
+    rs3 = random.sample(list1000, 100)
+
+    assert (rs1 == rs2) & (rs1 != rs3)
+
+
+@pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234])
+def test_cnn(input):
```
Comment (louisfh) [inline tests/test_utils.py:229]: Unclear what this is testing. Comment or make test name more explicit.

**Review by louisfh — APPROVED** [review]
Looks good - made a few small comments.

File: opensoundscape/ml/cnn.py (hunk 1 above)
Comment (sammlapp) [reply opensoundscape/ml/cnn.py:671]: we no longer use or report batch-level predictions and scores during the training loop, we just report the loss averaged

File: tests/test_utils.py (hunk 4 above)
Comment (sammlapp) [reply tests/test_utils.py:229]: good call, this test is actually testing set_seed, and making sure cnn.init results in consistent weights

File: opensoundscape/utils.py (hunk 3 above)
Comment (sammlapp) [reply opensoundscape/utils.py:209]: added test

File: opensoundscape/preprocess/actions.py (hunk 2 above)
Comment (sammlapp) [reply opensoundscape/preprocess/actions.py:66]: was just leftover code from refactor, we now create a new Series rather than updating the old one
